File: Backend/Models/Zingsnis.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    [Table("zingsnis")]
    public class Zingsnis
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [Column("komentaras")]
        public string Komentaras { get; set; } = null!;

        [Column("completed_at")]
        public DateTime? CompletedAt { get; set; }

        [Required]
        [Column("irasas_testas_id")]
        public int TestasIrasasId { get; set; }

        [Required]
        [Column("zingsnis_template_id")]
        public int ZingsnisTemplateId { get; set; }

        [Required]
        [Column("completed_by_user_id")]
        public Guid CompletedByUserId { get; set; }

        [JsonIgnore]
        public TestasIrasas? TestasIrasas { get; set; }

        [JsonIgnore]
        public ZingsnisTemplate? ZingsnisTemplate { get; set; }

        [JsonIgnore]
        public Naudotojas? CompletedByUser { get; set; }

        [NotMapped]
        public bool Pabaigtas { get; set; }

    }
}

File: frontend/lib/models/zingsnis.dart
class Zingsnis {
  final int id;
  String komentaras;
  DateTime? completedAt;
  int testasIrasasId;
  int zingsnisTemplateId;
  String completedByUserId;

  Zingsnis({
    required this.id,
    required this.komentaras,
    required this.completedAt,
    required this.testasIrasasId,
    required this.zingsnisTemplateId,
    required this.completedByUserId,
  });

  factory Zingsnis.fromJson(Map<String, dynamic> j) {
    final completedRaw = j['completedAt'] ?? j['completed_at'];
    DateTime? completed;
    if (completedRaw is String && completedRaw.isNotEmpty) {
      completed = DateTime.tryParse(completedRaw)?.toLocal();
    }

    return Zingsnis(
      id: j['id'] as int,
      komentaras: (j['komentaras'] ?? '') as String,
      completedAt: completed,
      testasIrasasId:
          (j['testasIrasasId'] ?? j['irasas_testas_id'] ?? 0) as int,
      zingsnisTemplateId:
          (j['zingsnisTemplateId'] ?? j['zingsnis_template_id'] ?? 0) as int,
      completedByUserId:
          (j['completedByUserId'] ?? j['completed_by_user_id'] ?? '') as String,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'komentaras': komentaras,
        'completedAt': completedAt?.toUtc().toIso8601String(),
        'testasIrasasId': testasIrasasId,
        'zingsnisTemplateId': zingsnisTemplateId,
        'completedByUserId': completedByUserId,
      };
}
